### DISCUSSION TITLE
how can I get the weifgt of a Bitvec var

### DISCUSSION THREAD

**Discussion by 1214465750** [Z3Prover/z3#6377] [Q&A]
for example
c is a Bitvec('c',8)
c=0b11010001
I want to get the number of 1 appeard in c
here the weight of c is 4
so how can I do this

**Comment by NikolajBjorner** [comment]
You can create an expression that returns number of 1's.
I think Marijn Heule used this as an example for homework in a class: you can create an optimized expression using bit-fiddling (see hacker's delight) or a straight-forward expression that extracts each bit and adds them up after padding.
